definitions.h

```diff
@@ -327,6 +385,82 @@ class Attributes
         }
 };
 
+/****************************************
+ * The operator overloading from the
+ * Matrix class are being defined
+ ***************************************/
+void Matrix::operator*=(const Matrix & rhs)
+{
+    // Initalizing variables
+    double temp[4][4];
+    double tempVal = 0;
+
+    // Checks if we can do that matrix math in the first place
+    if (rhs.cols != getRows())
+        throw "ERROR: Invalid sizes\n";
+
+    // Loop through the number of rows in the RHS
+    for (int x = 0; x < rhs.rows; x++)
+    {
+        // Loop through the number of cols in LHS
+        for (int y = 0; y < getCols(); y++)
+        {
+            double transfer[4];
+
+            //Flatten the LHS cols into a row for maths
+            for (int i = 0; i < getRows(); i++)
+                transfer[i] = this->data[i][y];
+
+            // Multiplies the rows and colloums
+            for (int i = 0; i < rhs.cols; i++)
+                tempVal += rhs.data[x][i] * transfer[i];
+
+            temp[x][y] = tempVal;
+        }
+    }
+    
+    // Changes the rows for the matrix
+    setRows(rhs.rows);
+
+    // Replaces the old matrix
+    for (int x = 0; x < getRows(); x++)
+        for (int y = 0; y < getCols(); y++)
+            this->data[x][y] = temp[x][y];
+}
+
+/******************************************************
+ * MATRIX MULT OVERLOADED OPERATOR
+ * Multiplies the two matricies together
+ *****************************************************/ 
+Vertex operator * (const Vertex & lhs, const Matrix & rhs)
```

Comment:
I see how you did your operator overload.  Good job.  In particular, good job on using a Vertex for the lhs.